src/SSVOpenHexagon/Global/Config.cpp

```diff
@@ -646,14 +646,17 @@ void setSaveLocalBestReplayToFile(bool mX)
     return fullscreen();
 }
 
-[[nodiscard, gnu::const]] float getVersion()
+[[nodiscard, gnu::const]] Version getVersion()
 {
-    return 2.03f;
+    return GAME_VERSION;
 }
 
 [[nodiscard, gnu::const]] const char* getVersionString()
 {
-    return "2.03";
+    return (std::to_string(GAME_VERSION.major) + "." +
+            std::to_string(GAME_VERSION.minor) + "." +
+            std::to_string(GAME_VERSION.micro))
+        .c_str();
```

Comment:
This is **undefined behavior**, as you're creating a temporary string and then returning `.c_str()`, which will be a pointer to some memory location which will immediately become invalid after returning from the function.

I would probably change this function to:

```cpp
[[nodiscard]] const std::string& getVersionString()
{
    static std::string result{
        std::to_string(GAME_VERSION.major) + "." +
        std::to_string(GAME_VERSION.minor) + "." +
        std::to_string(GAME_VERSION.micro)
    };

    return result;
}
```

The `static` guarantees that the string will be constructed once in a thread-safe way, and that will live for the entire duration of the program.

Reply:
You can also then remove the `std::string{...}` constructions from other places like in `u_getVersion`.

Reply:
I was just confused because the type was changed from std::string to a const char*, which behaves differently from a string. I was trying to use the .c_str() to try and get a character pointer from a string, but I guess this works better than const char*

Reply:
Alright rewrote the implementation using this suggestion, and made necessary changes in other parts of the code.